repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala

```diff
@@ -20,22 +20,210 @@
+    * I hope not, because that poops on what i'm trying to do with the cache at the moment.
+    *
+    * @param issuer the issuer
+    * @return the compiled xslt that represents the policy
+    * @throws SamlPolicyException for so many reasons
+    */
+  def getPolicy(issuer: String): XsltExecutable = ???
+
+  /**
+    * Applies the policy to the saml response.
+    *
+    * @param document the parsed saml response
+    * @param policy the xslt translation
+    * @return the translated document
+    * @throws SamlPolicyException if the translation fails
+    */
+  def translateResponse(document: Document, policy: XsltExecutable): Document = ???
+
+  /**
+    * Signs the saml response.
+    *
+    * @param document the translated saml response
+    * @return a signed saml response
+    * @throws SamlPolicyException if the signing fails
+    */
+  def signResponse(document: Document): Document = ???
+
+  /**
+    * Converts the saml response document into a servlet input stream for passing down the filter chain.
+    *
+    * @param document the final saml response
+    * @return the sinput stream that can be wrapped and passed along
+    */
+  def convertDocumentToStream(document: Document): ServletInputStream = ???
+
+  /**
+    * This is super vague because we don't know what it looks like yet.
+    *
+    * @return
+    */
+  def doResponseStuff() = ???
+
+  /**
+    * Evict from the cache as appropriate.
+    *
+    * @param atomEntry A {@link String} representation of an Atom entry. Note that Atom entries are XML elements,
+    */
+  override def onNewAtomEntry(atomEntry: String): Unit = ???
+
+  /**
+    * I suspect we don't care, but i could be wrong.
+    *
+    * @param event A value representing the new lifecycle stage of the system associated with the Feed that this
+    */
+  override def onLifecycleEvent(event: LifecycleEvents): Unit = ???
+
+  /**
+    * Stores the configuration and marks the filter as initialized.
+    * I'm going to have to initialize the cache and atom feed listener here, but it's not in the critical path for the moment.
+    *
+    * @param newConfiguration
+    */
+  override def configurationUpdated(newConfiguration: SamlPolicyConfig): Unit = {
+    val requestedFeedId = Option(newConfiguration.getPolicyAcquisition.getCache.getAtomFeedId)
+    if (feedId.nonEmpty && (requestedFeedId != Option(configuration.getPolicyAcquisition.getCache.getAtomFeedId))) {
+      atomFeedService.unregisterListener(feedId.get)
+      feedId = None
+    }
+
+    if (feedId.isEmpty && requestedFeedId.nonEmpty) {
+      feedId = Option(atomFeedService.registerListener(requestedFeedId.get, this))
+    }
+
+    if (Option(configuration).map(_.getPolicyAcquisition.getCache.getTtl) != Option(newConfiguration.getPolicyAcquisition.getCache.getTtl)) {
+      cache = CacheBuilder.newBuilder()
```

Comment:
Anybody got a good idea on how to test this? My best ideas are to expose the cache reflectively to check if the if is working correctly, and do an inheritance of this class to replace getPolicy to test that the ttl and get are working. These are terrible, but as i said my "best" ideas.